lib/spack/spack/platforms/cray.py

```diff
@@ -16,20 +18,33 @@
 
 def _get_modules_in_modulecmd_output(output):
     '''Return list of valid modules parsed from modulecmd output string.'''
-    return [i for i in output.splitlines()
+    return [i for i in re.split(r'  |\n', output)
             if len(i.split()) == 1]
 
 
-def _fill_craype_targets_from_modules(targets, modules):
+def _fill_craype_target_names_from_modules(targets, modules):
     '''Extend CrayPE CPU targets list with those found in list of modules.'''
     # Craype- module prefixes that are not valid CPU targets.
-    non_targets = ('hugepages', 'network', 'target', 'accel', 'xtpe')
+    non_targets = (
+        'hugepages', 'network', 'target', 'accel', 'xtpe', 'dl-plugin')
```

Comment:
This also seems to be fine, and unit testable. I'll change it to actually return the list so that it does not need to be passed as an argument.